File: Algorithm I/2. Two Pointers - Day 2, 3, 4, 5/P344.cpp
// Date: Mon, Sep 27, 2021
// 344. Reverse String (https://leetcode.com/problems/reverse-string/)

// time complexity O(n)
// space complexity O(1)
void reverseString(vector<char>& s) {
	for(int left = 0, right = s.size() - 1; left < right; left++, right--)
		swap(s[left], s[right]);
}

File: LeetCoding-Challenge/April-2022/P344.java
package leetcodingchallenge.april2022;

/**
 * Date: April 01, 2022
 * 344. Reverse String
 * https://leetcode.com/problems/reverse-string/
 */

public class P344 {
    // time: O(n)
    // space: O(1)
    public void reverseString(char[] s) {
        int left = 0, right = s.length - 1;
        while (left < right) {
            char tmp = s[left];
            s[left++] = s[right];
            s[right--] = tmp;
        }
    }
}
